test/PandoraInterface.cxx

```diff
@@ -336,14 +368,462 @@ void ProcessEvents(const Parameters &parameters, const Pandora *const pPrimaryPa
+
+    // Check if the start and end points are inside the voxelisation region
+    const bool inStart = grid.inside(start);
+    const bool inStop = grid.inside(stop);
+
+    if (inStart && inStop)
+    {
+        //std::cout << "Case 1: Start and end points are inside boundary" << std::endl;
+        point1 = start;
+        point2 = stop;
+    }
+    else if (!inStart && !inStop)
+    {
+        //std::cout << "Case 2: Start and end points are outside boundary" << std::endl;
+        bool ok = grid.intersect(ray, t0, t1);
+        if (ok)
+        {
+            point1 = ray.getPoint(t0);
+            point2 = ray.getPoint(t1);
+            //std::cout << "Boundary points: " << point1 << " and " << point2 << std::endl;
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary limits." << std::endl;
+            return currentVoxelList;
+        }
+    }
+    else if (inStart && !inStop)
+    {
+        //std::cout << "Case 3: Start inside boundary" << std::endl;
+        point1 = start;
+        bool ok = grid.intersect(ray, t0, t1);
+        if (ok)
+        {
+            point2 = ray.getPoint(t1);
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary" << std::endl;
+            return currentVoxelList;
+        }
+    }
+    else if (!inStart && inStop)
+    {
+        //std::cout << "Case 4: End inside boundary" << std::endl;
+        point2 = stop;
+        bool ok = grid.intersect(ray, t0, t1);
+        if (ok)
+        {
+            point1 = ray.getPoint(t0);
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary" << std::endl;
+            return currentVoxelList;
+        }
+    }
+
+    // Now create voxels between point1 and point2.
+    // Ray direction will be the same, but update starting point
+    ray.updateOrigin(point1);
+
+    const double epsilon(1e-3);
+    bool shuffle(true);
+
+    // Keep track of total voxel path length and energy (so far)
+    double totalPath(0.0), ETot(0.0);
+    int loop(0);
+
+    while (shuffle)
+    {
+
+        // Get point along path to define voxel bin (bottom corner)
+        pandora::CartesianVector voxelPoint = ray.getPoint(epsilon);
+
+        // Grid 3d bin containing this point; 4th element is the total bin number
+        std::array<long, 4> gridBins = grid.getBinIndices(voxelPoint);
+        long voxelID = gridBins[3];
+
+        long xBin = gridBins[0];
+        long yBin = gridBins[1];
+        long zBin = gridBins[2];
+
+        // Voxel bottom and top corners
+        const pandora::CartesianVector voxBot = grid.getPoint(xBin, yBin, zBin);
+        const pandora::CartesianVector voxTop = grid.getPoint(xBin + 1, yBin + 1, zBin + 1);
+
+        // Voxel box
+        const LArBox vBox(voxBot, voxTop);
+        // Get ray intersections with this box
+        bool result = vBox.intersect(ray, t0, t1);
+
+        if (!result)
```

Comment:
Result only used on this one line, so just put the `vBox.intersect` call in the if statement

Reply:
Decided to move this to be part of the if statement as suggested, since it also updates t0 and t1.